device/keypair.go

```diff
@@ -33,15 +33,15 @@ type Keypair struct {
 }
 
 type Keypairs struct {
-	sync.RWMutex
+	sync.Mutex
 	current  *Keypair
 	previous *Keypair
-	next     atomic.Pointer[Keypair]
+	next     *Keypair
 }
 
 func (kp *Keypairs) Current() *Keypair {
-	kp.RLock()
-	defer kp.RUnlock()
+	kp.Lock()
+	defer kp.Unlock()
```

Comment:
What's all this about? Why?

Reply:
And why isn't this a separate pull or commit? What's the motivation for doing this together with the rest exactly?

Reply:
It was, I sent it in September: https://github.com/WireGuard/wireguard-go/pull/58

Per our discussion, we folded our core set of performance changes into a summary commit.

Reply:
That wasn't what I intended to be the conclusion of our discussion. Rather, I'm okay if you want all of the GRO/GSO sendmmsg magic to be a single commit, if you view that as a continuous thing that's going to be a massive PITA to separate into a series LKML-style. But if you have new, additional performance optimizations, or anything else, submit those separately, with their own justification and review (as you did with #58). To the extent it's easy to pull things apart, do it. To the extent that it hampers productivity for little gain, it's find to have it smushed.

Reply:
I reverted this in https://github.com/WireGuard/wireguard-go/pull/64/commits/1c28dcd8c4532ba32b73c18ceb59b93a60d82b51. Discussion can pick up in #58.